fvm/transactionInvoker.go

```diff
@@ -69,28 +74,33 @@ func (i *TransactionInvoker) Process(
 		if sth.NumNestedTransactions() > 1 {
 			err := sth.RestartNestedTransaction(nestedTxnId)
 			if err != nil {
-				processErr = fmt.Errorf(
-					"cannot restart nested transaction: %w",
-					err,
+				processErr = errors.FirstOrFailure(
```

Comment:
It's not clear what we do when hitting failure.

It seems `processErr` could be a failure, in that case, do we still want to run `RestartNestedTransaction` in this defer call? What does `RestartNestedTransaction` do?

Shouldn't we stop everything if running into failure (fatal error)?

Reply:
restart nested transaction drops the nested transaction's deltas, but preserve the spock (and meter) bookkeeping.  Normally, if we didn't care about spock, we could just abort/discard the nested transaction.